Source/Core/Common/CommonPaths.h

```diff
@@ -50,6 +50,7 @@
 #define GAMESETTINGS_DIR "GameSettings"
 #define MAPS_DIR "Maps"
 #define CACHE_DIR "Cache"
+#define STATE_DIR "State"
```

Comment:
It looks like you're making a `State` sub-directory on Windows and macOS? That's maybe odd?
Should `%localappdata%` be used on Windows? `State` is maybe an ambiguous name here, with "save states" being a thing? Does macOS have an equivalent or should the files maybe just stay where they were?